plugins/action/common/prepare_plugins/prep_001_list_defaults.py

```diff
@@ -102,26 +102,28 @@ def prepare(self):
 
         # Check vxlan.topology.switches[index].freeforms list elements
         list_index = 0
-        for switch in self.model_data['vxlan']['topology']['switches']:
-            dm_check = data_model_key_check(switch, ['freeforms'])
-            if 'freeforms' in dm_check['keys_not_found'] or \
-               'freeforms' in dm_check['keys_no_data']:
-                self.model_data['vxlan']['topology']['switches'][list_index]['freeforms'] = []
+        if self.model_data['vxlan'].get('topology', None) is not None:
```

Comment:
I had planned to address the fact that there might not be topology data in a slightly different way.  The changes are not in the `develop` branch yet but they are in our `msd` branch.  Instead of the if check here, earlier I add the following check and set the topology data to an empty dict.  That way we still build out the entire structure even if it's empty.

https://github.com/netascode/ansible-dc-vxlan/blob/msd/plugins/action/common/prepare_plugins/prep_001_list_defaults.py#L76